### ISSUE TITLE
mv can partially move directories

### ISSUE THREAD

**Issue by gsuuon** [nushell/nushell#9041]
### Describe the bug

`mv` might do a partial move if there are failures, leaving parts of a folder's contents in both the old and new directory

### How to reproduce

1. create git repository
2. open in second terminal
3. move the repository
4. .git files are partially moved
5. old directory `git log` shows not a git repo
6. new directory `git log` shows no commits

### Expected behavior

`mv` should either work completely or fail completely without any filesystem changes 

### Screenshots

https://user-images.githubusercontent.com/6422188/235062418-2447311f-8e31-48f9-96d4-4f17aa539a7a.mp4

https://user-images.githubusercontent.com/6422188/235062447-3f2f9201-a02a-40d6-bb53-cbde53ef666b.mp4

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.77.0                                   |
| branch             |                                          |
| commit_hash        | fd09609b44b53d41ac7929e90708cdeba7bfb571 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)      |
| rust_channel       | 1.66.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)      |
| build_time         | 2023-03-14 20:53:18 +00:00               |
| build_rust_channel | release                                  |
| features           | default, zip                             |
| installed_plugins  |                                          |

**Comment by fdncred** [comment]
since you're on 0.77, can you try again with 0.79?

**Comment by gsuuon** [comment]
Here's where the issue is (through `fs_extra`)

https://github.com/nushell/nushell/blob/56e35fc3f99bc68fb160c3ee975290995613f4b8/crates/nu-command/src/filesystem/mv.rs#L314-L352

Potential solutions would be:
1. Only attempt rename on Windows, no copy/delete fallback
2. Only attempt copy/delete fallback if the from and to are on different fs or devices
3. Fix fs_extra such that it only attempts the copy/delete if all files can successfully be copied/deleted
4. Maybe coreutils correctly deals with file handling errors on Windows?

**Comment by fdncred** [comment]
I think this problem with fs_extra is also hurting coreutils. I think @dmatos2012 is currently trying to work around it.

**Comment by dmatos2012** [comment]
Yeah currently I am working on porting the `uutils` to nushell, and it was sort of ready till we hit a brick with moving/renaming consistently across OS.  Could you run the same but with `uutils mv` to see if you get an error, or it succeeds? @gsuuon

**Comment by gsuuon** [comment]
Looks like I get the same behavior with uutils mv:

https://github.com/nushell/nushell/assets/6422188/db6abf13-c6a3-4035-997a-e5d36adfee64

Some of the .git files are in use so fail but only after partially moving other files.
